tests/calculations/test_pw.py

```diff
@@ -30,3 +34,36 @@ def test_pw_default(fixture_sandbox, generate_calc_job, generate_inputs_pw, file
     # Checks on the files written to the sandbox folder as raw input
     assert sorted(fixture_sandbox.get_content_list()) == sorted(['aiida.in', 'pseudo', 'out'])
     file_regression.check(input_written, encoding='utf-8', extension='.in')
+
+
+def test_pw_parallelization_inputs(fixture_sandbox, generate_calc_job, generate_inputs_pw):
+    """Test that the parallelization settings are set correctly in the commandline params."""
+    Int = DataFactory('int')  # pylint: disable=invalid-name
+    entry_point_name = 'quantumespresso.pw'
+
+    inputs = generate_inputs_pw()
+    inputs['parallelization'] = {'nimage': Int(1), 'npool': Int(4), 'nband': Int(2), 'ntg': Int(3), 'ndiag': Int(12)}
+    calc_info = generate_calc_job(fixture_sandbox, entry_point_name, inputs)
+
+    cmdline_params = ['-nimage', '1', '-npool', '4', '-nband', '2', '-ntg', '3', '-ndiag', '12', '-in', 'aiida.in']
+
+    # Check that the command-line parameters are as expected.
+    assert calc_info.cmdline_params == cmdline_params
+
+
+@pytest.mark.parametrize(
+    'flag_name', ['nimage', 'ni', 'npool', 'nk', 'nband', 'nb', 'ntg', 'nt', 'northo', 'ndiag', 'nd']
+)
+def test_pw_parallelization_deprecation(fixture_sandbox, generate_calc_job, generate_inputs_pw, flag_name):
+    """Test the deprecation warning on specifying parallelization flags manually.
+
+    Test that passing parallelization flags in the `settings['CMDLINE']
+    emits a `DeprecationWarning`.
+    """
+    Dict = DataFactory('dict')  # pylint: disable=invalid-name
+    entry_point_name = 'quantumespresso.pw'
+
+    inputs = generate_inputs_pw()
+    inputs['settings'] = Dict(dict={'CMDLINE': ['-{}'.format(flag_name), '2']})
+    with pytest.warns(DeprecationWarning):
+        generate_calc_job(fixture_sandbox, entry_point_name, inputs)
```

Comment:
maybe still check here that the value of the settings is kept and not overridden?

Reply:
Done!